sensorhub-webui-core/src/main/java/org/sensorhub/ui/ReadmePanel.java

```diff
@@ -0,0 +1,164 @@
+            "<li><p>Your node&#39;s <code>build.gradle</code> file (the outermost one) includes the following:\n" +
+            "   </p>\n" +
+            "<pre><code><span class=\"hljs-keyword\">allprojects</span> {\n" +
+            "    version = oshCoreVersion\n" +
+            "    tasks.register(<span class=\"hljs-string\">'copyReadme'</span>, <span class=\"hljs-keyword\">Copy</span>) {\n" +
+            "        <span class=\"hljs-keyword\">from</span> <span class=\"hljs-string\">\"${projectDir}/README.md\"</span>\n" +
+            "        <span class=\"hljs-keyword\">into</span> <span class=\"hljs-string\">\"${buildDir}/resources/main\"</span>\n" +
+            "        onlyIf { <span class=\"hljs-keyword\">file</span>(<span class=\"hljs-string\">\"${projectDir}/README.md\"</span>).exists() }\n" +
+            "    }\n" +
+            "}\n" +
+            "\n" +
+            "<span class=\"hljs-keyword\">subprojects</span> {\n" +
+            "    <span class=\"hljs-comment\">// inject all repositories from included builds if any</span>\n" +
+            "    <span class=\"hljs-keyword\">repositories</span>.addAll(rootProject.<span class=\"hljs-keyword\">repositories</span>)\n" +
+            "    plugins.withType(JavaPlugin) {\n" +
+            "        processResources {\n" +
+            "            dependsOn copyReadme\n" +
+            "        }\n" +
+            "    }\n" +
+            "}\n" +
+            "</code></pre></li>\n" +
+            "</ul>";
+
+    // This determines which tab is visible
+    // Hack needed for desired accordion behavior in this older version of Vaadin
+    private boolean visibleTab = false;
+
+    @JavaScript({"vaadin://js/jquery.min.js", "vaadin://js/lodash.min.js", "vaadin://js/backbone.min.js", "vaadin://js/joint.js", "vaadin://js/marked.min.js", "vaadin://js/readme.js"})
+    public class ReadmeJS extends AbstractJavaScriptComponent {
+        private InputStream readmeIs;
+        private static final Logger logger = LoggerFactory.getLogger(ReadmePanel.class);
+        private boolean hasContent = false;
+
+        public static class ReadmeState extends JavaScriptComponentState {
+            public String readmeText;
+        }
+
+        private ReadmeJS(final MyBeanItem<ModuleConfig> beanItem) {
+
+            try {
+                // Get the JAR file location from the protection domain
+                CodeSource codeSource = beanItem.getBean().getClass().getProtectionDomain().getCodeSource();
+                if (codeSource != null) {
+
+                    // Convert JAR URL to a file path, navigate to the build directory
+                    File jarFile = new File(codeSource.getLocation().toURI());
+                    File buildDir = jarFile.getParentFile().getParentFile();
+
+                    // Look for README.md in the resources directory
+                    File readmeFile = new File(buildDir, "resources/main/README.md");
```

Comment:
Sorry I merged too fast. This cannot work because the build directory is not supposed to exist at runtime. It works in your IDE but not if the app is packaged into a JAR.

The right place to put this is inside `src/resources/main/{full/driver/package/hierarchy}/README.md` and then it will be packaged into the JAR.

Then from code you can get the resource as an inputstream by doing:
`beanItem.getBean().getClass().getResourceAsStream("README.md")`